$NetBSD: patch-af,v 1.4 2011/03/22 23:31:04 tez Exp $

Add --enable-pkgsrc-libtool option
Use $(ROOT_USER) in place of root

--- config/pre.in.orig	2010-01-21 16:49:01.000000000 -0600
+++ config/pre.in	2010-12-12 17:39:24.427787200 -0600
@@ -170,6 +170,7 @@
 THREAD_LINKOPTS = $(PTHREAD_CFLAGS) $(PTHREAD_LIBS)
 CPPFLAGS = @CPPFLAGS@
 DEFS = @DEFS@
+LIBTOOL = @LIBTOOL@
 CC = @CC@
 CXX = @CXX@
 LD = $(PURE) @LD@
@@ -183,11 +184,12 @@
 
 INSTALL=@INSTALL@
 INSTALL_STRIP=
-INSTALL_PROGRAM=@INSTALL_PROGRAM@ $(INSTALL_STRIP)
-INSTALL_SCRIPT=@INSTALL_PROGRAM@
+INSTALL_LIBTOOL=@INSTALL_LIBTOOL@
+INSTALL_PROGRAM=$(INSTALL_LIBTOOL) @INSTALL_PROGRAM@ $(INSTALL_STRIP)
+INSTALL_SCRIPT=@INSTALL_SCRIPT@
 INSTALL_DATA=@INSTALL_DATA@
 INSTALL_SHLIB=@INSTALL_SHLIB@
-INSTALL_SETUID=$(INSTALL) $(INSTALL_STRIP) -m 4755 -o root
+INSTALL_SETUID=$(INSTALL_LIBTOOL) $(INSTALL) $(INSTALL_STRIP) -m 4555 -o $(ROOT_USER)
 ## This is needed because autoconf will sometimes define @exec_prefix@ to be
 ## ${prefix}.
 prefix=@prefix@
@@ -313,6 +315,10 @@
 # depending on whether we're building with shared libraries.
 DEPLIBEXT=@DEPLIBEXT@
 
+# this gets set by configure to either $(STLIBEXT) or $(LALIBEXT),
+# depending on whether we're building libtool archives.
+DEPSTLIBEXT=@DEPSTLIBEXT@
+
 KDB5_PLUGIN_DEPLIBS = @KDB5_PLUGIN_DEPLIBS@
 KDB5_PLUGIN_LIBS = @KDB5_PLUGIN_LIBS@
 
@@ -329,12 +335,12 @@
 SUPPORT_LIBNAME=krb5support
 SUPPORT_DEPLIB	= $(TOPLIBD)/lib$(SUPPORT_LIBNAME)$(DEPLIBEXT)
 
-# These are forced to use ".a" as an extension because they're never
+# These are forced to use "@DEPSTLIBEXT" as an extension because they're never
 # built shared.
 SS_DEPLIB	= $(SS_DEPLIB-@SS_VERSION@)
-SS_DEPLIB-k5	= $(TOPLIBD)/libss.a
+SS_DEPLIB-k5	= $(TOPLIBD)/libss$(DEPSTLIBEXT)
 SS_DEPLIB-sys	=
-APPUTILS_DEPLIB	= $(TOPLIBD)/libapputils.a
+APPUTILS_DEPLIB	= $(TOPLIBD)/libapputils$(DEPSTLIBEXT)
 
 KRB5_BASE_DEPLIBS	= $(KRB5_DEPLIB) $(CRYPTO_DEPLIB) $(COM_ERR_DEPLIB) $(SUPPORT_DEPLIB)
 KDB5_DEPLIBS		= $(KDB5_DEPLIB) $(KDB5_PLUGIN_DEPLIBS)
@@ -364,7 +370,7 @@
 
 SS_LIB		= $(SS_LIB-@SS_VERSION@)
 SS_LIB-sys	= @SS_LIB@
-SS_LIB-k5	= $(TOPLIBD)/libss.a
+SS_LIB-k5	= $(TOPLIBD)/libss$(DEPSTLIBEXT)
 KDB5_LIB	= -lkdb5 $(KDB5_PLUGIN_LIBS)
 
 DL_LIB		= @DL_LIB@
@@ -509,6 +515,9 @@
 DYNOBJ_EXPDEPS=@DYNOBJ_EXPDEPS@
 DYNOBJ_EXPFLAGS=@DYNOBJ_EXPFLAGS@
 
+# usually .la
+LALIBEXT=@LALIBEXT@
+
 # File with symbol names to be exported, both functions and data,
 # currently not distinguished.
 SHLIB_EXPORT_FILE=$(srcdir)/$(LIBPREFIX)$(LIBBASE).exports
@@ -530,6 +539,7 @@
 # flags for explicit libraries depending on this one,
 # e.g. "$(SHLIB_RPATH_FLAGS) $(SHLIB_SHLIB_DIRFLAGS) $(SHLIB_EXPLIBS)"
 SHLIB_EXPFLAGS=@SHLIB_EXPFLAGS@
+LALIB_EXPFLAGS=@LALIB_EXPFLAGS@
 
 ## Parameters to be set by configure for use in libobj.in:
 
@@ -541,10 +551,15 @@
 # the suffix substitution will break on some platforms!
 SHLIBOBJS=$(STLIBOBJS:.o=@SHOBJEXT@)
 PFLIBOBJS=$(STLIBOBJS:.o=@PFOBJEXT@)
+LALIBOBJS=$(STLIBOBJS:.o=@LAOBJEXT@)
 
 # "$(CC) -G", "$(LD) -Bshareable", etc.
 LDCOMBINE=@LDCOMBINE@
 
+# Args to tack on the tail of LIBTOOL to generate versioned, installable
+# libtool archives.
+LIBTOOL_TAIL=@LIBTOOL_TAIL@
+
 #
 # rules to make various types of object files
 #
